### PR TITLE
Opaque bool

### PR DESCRIPTION
**Pull request by KilianVounckx**
Implement #3331 

Big thanks to @ayazhafiz for all the help with this!

### CODE REVIEW

File: crates/compiler/test_mono/generated/if_multi_branch.txt
Comment (ayazhafiz) [inline crates/compiler/test_mono/generated/if_multi_branch.txt:7]: This is **begging** for inlining ;)

A note aside: we might want to special case this in mono so that programs that don't actually use anything in `Bool` other than true/false don't have to enter into specializing `Bool` (which is a marginal cost, but a cost nonetheless, because a thread has to take the time to generate these two functions, but we know we can unconditionally inline them). That doesn't need to be done in this PR, though.

**Comment by ayazhafiz** [comment]
@KilianVounckx can you sign your commits? Otherwise after CI passes, we will still be blocked on merging due to that.

**Review by ayazhafiz — APPROVED** [review]

**Comment by rtfeldman** [comment]
Outstanding, thank you so much @KilianVounckx and @ayazhafiz!!! 😍

**Pull request merged**
